lightning/src/ln/channelmanager.rs

```diff
@@ -3551,33 +3562,59 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> MessageSend
 				L::Target: Logger,
 {
 	fn get_and_clear_pending_msg_events(&self) -> Vec<MessageSendEvent> {
-		//TODO: This behavior should be documented. It's non-intuitive that we query
-		// ChannelMonitors when clearing other events.
-		self.process_pending_monitor_events();
+		let result = RefCell::new(Vec::new());
+		PersistenceNotifierGuard::optionally_notify(&self.total_consistency_lock, &self.persistence_notifier, || {
+			// TODO: This behavior should be documented. It's unintuitive that we query
+			// ChannelMonitors when clearing other events.
+			self.process_pending_monitor_events();
 
-		let mut ret = Vec::new();
-		let mut channel_state = self.channel_state.lock().unwrap();
-		mem::swap(&mut ret, &mut channel_state.pending_msg_events);
-		ret
+			let mut pending_events = Vec::new();
+			let mut channel_state = self.channel_state.lock().unwrap();
+			mem::swap(&mut pending_events, &mut channel_state.pending_msg_events);
+
+			if pending_events.is_empty() {
+				NotifyOption::SkipPersist
+			} else {
+				result.replace(pending_events);
+				NotifyOption::DoPersist
+			}
+		});
+		result.into_inner()
 	}
 }
 
+/// Provides events that must be periodically handled.
```

Comment:
These docs are collapsed by default on the ChannelManager page, beside four other trait implementations, making them basically invisible IMO. I don't know if there's a way to change rustdoc's behavior here, but I wonder if it makes sense to explicitly link to the known concrete implementations in the `EventsProvider` docs and say, eg, "eg see [\`ChannelManager::process_pending_events\`] and ..." which seems to handle opening the panel.